AirCasting/MicrophoneSession/MicrophoneManager.swift

```diff
@@ -21,86 +18,110 @@ class MicrophoneManager: ObservableObject {
-            AVFormatIDKey: NSNumber(value: kAudioFormatAppleLossless),
-            AVSampleRateKey: 44100.0,
-            AVNumberOfChannelsKey: 1,
-            AVEncoderAudioQualityKey: AVAudioQuality.min.rawValue
-        ]
-        
-        do {
-            try audioSession.setCategory(AVAudioSession.Category.playAndRecord)
-            try audioSession.setActive(true)
-            try recorder = AVAudioRecorder(url:url, settings: recordSettings)
             
-        } catch {
-            return
+            recorder.isMeteringEnabled = true
+            recorder.record()
+            isRecording = true
+            
+            levelTimer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(getMeasurement), userInfo: nil, repeats: true)
         }
-        
-        recorder.isMeteringEnabled = true
-        recorder.record()
-        isRecording = true
-        
-        levelTimer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(getMeasurement), userInfo: nil, repeats: true)
     }
     
     @objc func getMeasurement() {
+        if !recorder.isRecording {
+            recorder.record()
+        }
         recorder.updateMeters()
-        
         let level = recorder.averagePower(forChannel: 0)
+        createMeasurement(value: level)
+    }
+    
+    func stopRecording() {
+        levelTimer.invalidate()
+        isRecording = false
+        recorder.stop()
+        recorder = nil
+        session?.status = Int16(SessionStatus.FINISHED.rawValue)
+        do {
+            try context.save()
+        } catch {
+            print("Error for saving session when stopping mic recording: \(error)")
+        }
+    }
+    
+    @objc func handleInterruption(notification: Notification) {
+        guard let userInfo = notification.userInfo,
+            let typeValue = userInfo[AVAudioSessionInterruptionTypeKey] as? UInt,
+            let type = AVAudioSession.InterruptionType(rawValue: typeValue) else {
+                return
+        }
+
+        switch type {
+        case .began:
+            levelTimer.invalidate()
+            session?.status = Int16(SessionStatus.DISCONNETCED.rawValue)
+        case .ended:
+            levelTimer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(getMeasurement), userInfo: nil, repeats: true)
+        default: ()
+        }
+    }
+    
+    private
+    
+    func createMeasurement(value: Float) {
         let startingLocation = obtainCurrentLocation()
         
         let measurement = Measurement(context: self.context)
         
         measurement.latitude = startingLocation.latitude
         measurement.longitude = startingLocation.longitude
-        measurement.value = Double(level)
+        measurement.value = Double(value)
         measurement.time = Date()
         
         session?.endTime = measurement.time
-        session?.status = 0
+        session?.status = Int16(SessionStatus.RECORDING.rawValue)
```

Comment:
Use `session?.status = .RECORDING`. Also capitalized enum cases are not keeping the `Swift` standards.

Reply:
`session?.status = .RECORDING` doesn't work and I get an error "Type 'Int16' has no member 'RECORDING'"